lms/templates/navigation.html

```diff
@@ -21,6 +20,26 @@
 ## Provide a hook for themes to inject branding on top.
 <%block name="navigation_top" />
 
+## Add UI Toolkit components if using the Pattern Library
+% if uses_pattern_library:
+    <%block name="js_extra">
+    <%static:require_module module_name="js/header_factory" class_name="HeaderFactory">
+    HeaderFactory();
+    </%static:require_module>
+    </%block>
+% endif
+
+<%
+if uses_pattern_library:
```

Comment:
Thanks for pushing us to do the right thing, @dan-f. I've more aggressively changed the code to use the same classes for both v1 and v2 styles, and so I no longer have CSS class variables. I have left a couple of if/else conditionals for cases with radically different markup, but I hope you'll agree that they seem necessary. You can see the changes in this commit: https://github.com/edx/edx-platform/pull/12057/commits/b55b37726eb08ac7631ed9e912f500207f0b5a13

I've done some spot checking on my devstack and I'm going to update my sandbox now. Clearly this will require some more careful testing by the three of us to make sure that we're not breaking anything. My sense from analyzing the Sass is that the worst that will happen is that a header will be styled incorrectly which should be easy to spot.